backend/video/src/subtitle.py

```diff
@@ -35,85 +37,67 @@ def get_subtitles(self):
             # no subtitles
             return False
 
-        relevant_subtitles = []
-        for lang in self.languages:
-            user_sub = self._get_user_subtitles(lang)
-            if user_sub:
-                relevant_subtitles.append(user_sub)
-                continue
-
-            if self.video.config["downloads"]["subtitle_source"] == "auto":
-                auto_cap = self._get_auto_caption(lang)
-                if auto_cap:
-                    relevant_subtitles.append(auto_cap)
+        available_subtitles = self._get_all_subtitles("user")
+        if self.video.config["downloads"]["subtitle_source"] == "auto":
+            for lang, auto_cap in self._get_all_subtitles("auto"):
```

Comment:
self._get_all_subtitles("auto") returns a dictionary, do you mean:
```python
for lang, auto_cap in self._get_all_subtitles("auto").items()
```
?

Reply:
Good catch